vassal-app/src/main/java/VASSAL/counters/SendToLocation.java

```diff
@@ -388,8 +388,8 @@ public Command myKeyEvent(KeyStroke stroke) {
         mat = (Mat) Decorator.getDecorator(outer, Mat.class);
         if (mat != null) {
           final Point basePt = outer.getPosition();
-          contents = new ArrayList<>(mat.getContents());
-          offsets = new ArrayList<>(mat.getOffsets(basePt.x, basePt.y));
+          contents = mat.getContents();
```

Comment:
Oh, oh, oh. That's going to be bad for `contents`. This is what `Mat.getContents()` does:
```
 public List<GamePiece> getContents() {
    return contents;
  }
```
So it's _not_ returning a copy---it's returning the actual contents list held by the `Mat`, which whatever you've given it to can now twiddle at will. That's no bueno.

Reply:
I'm not actually twiddling it, but I guess obviously "someone could someday". 

Rather than keep guessing, maybe just let me know what you'd like this to say?

Reply:
Sorry. I thought you'd confirmed that `getContents()` was already returning a copy. Since it's not, you should do:
```
contents = new ArrayList<>(mat.getContents());
```
Alternatively, you could alter `getContents()` to return a copy. Whether that's a good idea depends on how much `getContents()` is called in contexts where you don't need a copy.

Reply:
Ah - okay since I can't imagine it ever being a good idea for any "recipient" of this to twiddle contents itself, I've changed getContents() to return a copy and changed one case where it would thus have been re-copying the copy to account for that. Yay copies.

Reply:
This is one of those things some languages resolve by having `const` and non-`const` versions of functions, so then you can make the `const` version return the real thing with a guarantee that it won't be modified.